client/src/pages/record.tsx

```diff
@@ -116,6 +121,12 @@ function RecordPage(props: {
       navigate("/");
     }
   }
+  function onEnd() {
```

Comment:
want to get away from fetch calls inside components. Looks like there's already a `useRecordState` hook. Can this be moved into there, e.g. `recordState.refetchMentor`?